bench/ecdsa.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright 2015-2023 Brian Smith.
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHORS DISCLAIM ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY
+// SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
+// OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+// CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+#![allow(missing_docs)]
+
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use ring::{
+    rand,
+    signature::{self, EcdsaKeyPair, EcdsaSigningAlgorithm, EcdsaVerificationAlgorithm, KeyPair},
+};
+
+static ALGORITHMS: &[(&str, &EcdsaSigningAlgorithm, &EcdsaVerificationAlgorithm)] = &[
+    (
+        "p256",
+        &signature::ECDSA_P256_SHA256_FIXED_SIGNING,
+        &signature::ECDSA_P256_SHA256_FIXED,
+    ),
+    (
+        "p384",
+        &signature::ECDSA_P384_SHA384_FIXED_SIGNING,
+        &signature::ECDSA_P384_SHA384_FIXED,
+    ),
+];
+
+fn sign(c: &mut Criterion) {
+    for (alg_name, alg, _) in ALGORITHMS {
+        c.bench_function(&format!("ecdsa_{alg_name}_sign"), |b| {
```

Comment:
We're going to end up with the "fixed" and non-fixed (ASN.1) benchmarks having the same benchmark name.

I suggest we just remove the `_FIXED_` variants from the bemchmarking to resolve this. I don't think we need to worry about the difference in performance between the two ways of formatting the signature and also the performance difference between the two is probably impossible to measure in a meaningful way.

Reply:
Not sure what do you mean, the name is whatever string I put there in the ALGORITHMS array. I also think padding performance is not critical, so I only put the FIXED variants.

Reply:
I don't remember what it looked like before but it is fine the way it is now.